### PR TITLE
use two different selenium version in same repository

### PR DESCRIPTION
**Pull request by TakeruUshio**
You can use two different selenium version in same repository. (selenium 3.x / selenium 4.x)

### check run features

<details>
<summary> use selenium3 </summary>

```shell
$ USE_HEADLESS=true BUNDLE_GEMFILE=gemfiles/selenium_3.gemfile bundle exec rspec spec/features/sample_use_selenium3_or_seleninum4.feature

サンプル
  use selenium3
I, [2019-09-27T10:40:17.859019 #45449]  INFO -- : [1] Finished step ブラウザを起動する (3.709 sec)
I, [2019-09-27T10:40:22.059960 #45449]  INFO -- : [2] Finished step 'https://travis-ci.org' にアクセスする (4.201 sec)
I, [2019-09-27T10:40:22.132266 #45449]  INFO -- : [3] Finished step 'Sign Up' ボタンが表示されていること (0.072 sec)
I, [2019-09-27T10:40:25.836993 #45449]  INFO -- : [4] Finished step Seleniumのブラウザを閉じる (3.705 sec)
    もしブラウザを起動する -> もし'https://travis-ci.org' にアクセスする -> ...(snip)...
  use selenium4
    もしブラウザを起動する -> もし'https://travis-ci.org' にアクセスする -> ...(snip)... (PENDING: No reason given)

Pending: (Failures listed here are expected and do not affect your suite's status)

  1) サンプル use selenium4 もしブラウザを起動する -> もし'https://travis-ci.org' にアクセスする -> ...(snip)...
     # No reason given
     # ./spec/features/sample_use_selenium3_or_seleninum4.feature:13


Top 2 slowest examples (11.69 seconds, 100.0% of total time):
  サンプル use selenium3 もしブラウザを起動する -> もし'https://travis-ci.org' にアクセスする -> ...(snip)...
    11.69 seconds ./spec/features/sample_use_selenium3_or_seleninum4.feature:6
  サンプル use selenium4 もしブラウザを起動する -> もし'https://travis-ci.org' にアクセスする -> ...(snip)...
    0.00024 seconds ./spec/features/sample_use_selenium3_or_seleninum4.feature:13

Finished in 11.69 seconds (files took 0.63658 seconds to load)
2 examples, 0 failures, 1 pending
```

</details>

<details>
<summary> use selenium4 </summary>

```shell
$ USE_HEADLESS=true BUNDLE_GEMFILE=gemfiles/selenium_4.gemfile bundle exec rspec spec/features/sample_use_selenium3_or_seleninum4.feature

サンプル
  use selenium3
    もしブラウザを起動する -> もし'https://travis-ci.org' にアクセスする -> ...(snip)... (PENDING: No reason given)
  use selenium4
I, [2019-09-27T10:41:05.254917 #45468]  INFO -- : [1] Finished step ブラウザを起動する (3.151 sec)
I, [2019-09-27T10:41:08.607203 #45468]  INFO -- : [2] Finished step 'https://travis-ci.org' にアクセスする (3.352 sec)
I, [2019-09-27T10:41:08.701314 #45468]  INFO -- : [3] Finished step 'Sign Up' ボタンが表示されていること (0.094 sec)
I, [2019-09-27T10:41:11.442771 #45468]  INFO -- : [4] Finished step Seleniumのブラウザを閉じる (2.741 sec)
    もしブラウザを起動する -> もし'https://travis-ci.org' にアクセスする -> ...(snip)...

Pending: (Failures listed here are expected and do not affect your suite's status)

  1) サンプル use selenium3 もしブラウザを起動する -> もし'https://travis-ci.org' にアクセスする -> ...(snip)...
     # No reason given
     # ./spec/features/sample_use_selenium3_or_seleninum4.feature:6


Top 2 slowest examples (9.34 seconds, 100.0% of total time):
  サンプル use selenium4 もしブラウザを起動する -> もし'https://travis-ci.org' にアクセスする -> ...(snip)...
    9.34 seconds ./spec/features/sample_use_selenium3_or_seleninum4.feature:13
  サンプル use selenium3 もしブラウザを起動する -> もし'https://travis-ci.org' にアクセスする -> ...(snip)...
    0.00016 seconds ./spec/features/sample_use_selenium3_or_seleninum4.feature:6

Finished in 9.34 seconds (files took 0.50737 seconds to load)
2 examples, 0 failures, 1 pending
```

</details>

### CODE REVIEW

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -31,6 +31,21 @@ def travis?
   # https://relishapp.com/rspec/rspec-core/docs/command-line/only-failures
   config.example_status_persistence_file_path = 'tmp/example_status.txt'
 
+  config.before do |scenario|
+    selenium_ver = ENV["BUNDLER_ORIG_BUNDLE_GEMFILE"]
```
Comment (minimum2scp) [inline spec/spec_helper.rb]: If you really want selenium-webdriver version, you can get it by

```ruby
Selenium::WebDriver::VERSION
```

and you can get its major version by

```ruby
selenium_major_ver = Gem::Version.new(Selenium::WebDriver::VERSION).segments[0]
```

or

```ruby
selenium_major_ver = Selenium::WebDriver::VERSION.split('.')[0]
```

File: spec/spec_helper.rb [hunk 2]
```diff
@@ -31,6 +31,22 @@ def travis?
...
+             when "3" then :selenium3
+             when "4" then :selenium4
+             else
+               raise "Selenium #{selenium_ver} version is not supported"
+             end
+
+    metadata = scenario.metadata
+    if metadata[:selenium3] || metadata[:selenium4]
+      skip unless metadata[s_sym]
+    end
```
Comment (minimum2scp) [inline spec/spec_helper.rb]: Umm... you want to run/skip examples on ... like this?

| metadata[:selenium3] | metadata[:selenium4] | s_sym      | Run or Skip |
|:--                   |:--                   |:--         |:--          |
| false                | false                | :selenium3 | Run         |
| false                | true                 | :selenium3 | Skip        |
| true                 | false                | :selenium3 | Run         |
| true                 | true                 | :selenium3 | Run         |
| false                | false                | :selenium4 | Run       |
| false                | true                 | :selenium4 | Run         |
| true                 | false                | :selenium4 | Skip        |
| true                 | true                 | :selenium4 | Run         |

Comment (TakeruUshio) [reply spec/spec_helper.rb]: yap!

Comment (minimum2scp) [reply spec/spec_helper.rb]: If so, then you can rewrite like this:

```
skip if (!metadata[:selenium3] && metadata[:selenium4] && s_sym == :selenium3) ||
        (metadata[:selenium3] && !metadata[:selenium4] && s_sym == :selenium4)
```

Comment (minimum2scp) [reply spec/spec_helper.rb]: OK, it smells but your code works as expected.

Comment (TakeruUshio) [reply spec/spec_helper.rb]: Fine thanks.

File: .travis.yml [hunk 3]
```diff
@@ -6,6 +6,10 @@ language: ruby
 rvm:
   - 2.6
 
+gemfile:
+  - gemfiles/selenium_3.gemfile
+  - gemfiles/selenium_4.gemfile
```
Comment (minimum2scp) [inline .travis.yml]: Why not append  `Gemfile`?

Comment (TakeruUshio) [reply .travis.yml]: oh.. I forgot!

File: gemfiles/selenium_3.gemfile [hunk 4]
```diff
@@ -0,0 +1,17 @@
+source 'https://rubygems.org'
+
+gem 'rspec', '3.8.0'
+
+group :test do
+  gem 'turnip'
+  gem 'representable', '3.0.4'
+  gem 'selenium-webdriver'
+  gem 'turnip_formatter'
+  gem 'activesupport', '< 5.0.0'
```
Comment (minimum2scp) [inline gemfiles/selenium_3.gemfile]:
```suggestion
  gem 'activesupport', '< 6.0.0'
```

File: gemfiles/selenium_4.gemfile [hunk 5]
```diff
@@ -0,0 +1,17 @@
+source 'https://rubygems.org'
+
+gem 'rspec', '3.8.0'
+
+group :test do
+  gem 'turnip'
+  gem 'representable', '3.0.4'
+  gem 'selenium-webdriver', '4.0.0.alpha3'
+  gem 'turnip_formatter'
+  gem 'activesupport', '< 5.0.0'
```
Comment (minimum2scp) [inline gemfiles/selenium_4.gemfile]:
```suggestion
  gem 'activesupport', '< 6.0.0'
```

**Pull request merged**